indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/Interfaces.kt

```diff
@@ -0,0 +1,237 @@
+     * @return                          created schema
+     */
+    fun createSchema(name: String, version: String, attributes: List<String>): Schema
+
+    /**
+     * Creates credential definition and stores it to ledger if not exists, else restores credential definition from ledger
+     *
+     * @param schemaId                  id of schema to create credential definition for
+     * @param enableRevocation          whether enable or disable revocation for this credential definition
+     *                                  (hint) turn this on by default, but just don't revoke credentials
+     *
+     * @return                          created credential definition
+     */
+    fun createCredentialDefinition(schemaId: String, enableRevocation: Boolean): CredentialDefinition
+
+    /**
+     * Creates revocation registry for credential definition if there's no one in ledger
+     * (usable only for those credential definition for which enableRevocation = true)
+     *
+     * @param credentialDefinitionId    credential definition id
+     * @param maxCredentialNumber       maximum number of credentials which can be issued for this credential definition
+     *                                  (example) driver agency can produce only 1000 driver licences per year
+     *
+     * @return                          created
+     */
+    fun createRevocationRegistry(credentialDefinitionId: String, maxCredentialNumber: Int = 5): RevocationRegistryInfo
+
+    /**
+     * Creates credential offer
+     *
+     * @param credentialDefinitionId    credential definition id
+     *
+     * @return                          created credential offer
+     */
+    fun createCredentialOffer(credentialDefinitionId: String): CredentialOffer
+
+    /**
+     * Issues credential by credential request. If revocation is enabled it will hold one of [maxCredentialNumber].
+     *
+     * @param credentialRequest         credential request and all reliable info
+     * @param proposal                  credential proposal
+     * @param offer                     credential offer
+     * @param revocationRegistryId      <optional> revocation registry definition ID
+     *
+     * @return                          credential and all reliable info
+     */
+    fun issueCredential(
+        credentialRequest: CredentialRequestInfo,
+        proposal: String,
+        offer: CredentialOffer,
+        revocationRegistryId: String?
+    ): CredentialInfo
+
+    /**
+     * Revokes previously issued credential
+     *
+     * @param revocationRegistryId      revocation registry definition id
+     * @param credentialRevocationId    revocation registry credential index
+     */
+    fun revokeCredential(revocationRegistryId: String, credentialRevocationId: String)
+}
+
+/**
+ * This entity is able to receive credentials and create proofs about them.
+ * Has read-only access to public ledger.
+ */
+interface IndyProver : IndyWalletHolder {
+    val defaultMasterSecretId: String
```

Comment:
`defaultMasterSecretId` was added to simplify initial POC, but technically it is a totally wrong idea, and I think it is time we start removing it. To keep this PR small I would remove `defaultMasterSecretId` from these interface and kept it as a file constant, for example. Adding `@Deprcated` wouldn't hurt ether